### ISSUE TITLE
Advise best method for each Label appearing "only once"

### ISSUE THREAD

**Issue by prabhatM** [facebookresearch/fastText#815]
Hi,

I have a labeled data set where each label appears only once and I have 3,00,000 such labels in a file.

Example :
__label__218072001 xxxxxx xxxx xxxxxxx xxxxxxxx
__label__466394009 xx xxxx xxxx xxxxx xxxxx
__label__163239008 xxxxxxx xxx xx xxxxx xxxxxxxxx
__label__412957009 xxxx xxxxxx xxxxxxx xxxx xxxxxx
__label__406945007 xxxxx xxxxx xxxxx xxxxxxxx xxxxxxxx
__label__129285002 xxxxx xxxxxx xxxx xxxxxxx xxxx xxx
__label__431848008 xxxx xxxxxx xxxxxxxxx xxxxxx xxxxxx

What is the best approach for a  "Supervised Learning" ?

I would prefer a bigram and the model must capture marginal spelling mistakes and a missing word in the text.

Since each label appears only once, I guess I must not divide the data into train and test. The whole data needs to be trained. What will be the ideal Learning Rate, epoch and loss for such a job ?

**Comment by giriannamalai** [comment]
You should use the whole train data instead of splitting. 

If you are doing multi label, then use one-vs-all
Else you can use his as loss function.

You can start learning rate from 0.05
If learning rate is small epoch should be higher. You can set 10000 epochs for this lr

Do this repeatedly and find which is match for yours

**Comment by prabhatM** [comment]
Does higher epoch increase the model size? I noticed so.

**Comment by giriannamalai** [comment]
It may. You could always quantize the model to reduce its size.

**Comment by prabhatM** [comment]
Thank you for your response

**Comment by giriannamalai** [comment]
Welcome...
If you found your answer, please close this thread.

**Comment by prabhatM** [comment]
Sure.
